File: linux_c/serial/receive.c
// FOR EDUCATIONAL PURPOSES ONLY, DO NOT USE IN PRODUCTION
// Source code is provided as-is without warranty, and disclaiming liability for damages resulting from using the source code.
// For simplicity there is no error handling or sanity check.

#include <stdio.h>
#include <unistd.h>
#include <fcntl.h>
#include <termios.h>
#include <string.h>

int main(){
  unsigned char buf[6];
  memset(buf, '\0', sizeof(buf));
  int fd = open("/dev/ttyUSB0", O_RDWR);
  int n = read(fd, &buf, sizeof(buf) - 1);
  printf("Bytes read: %d\n", n);
  printf("%s\n",buf);
  close(fd);
  return 0;
}

File: python3/socket/raw/receive.py
# FOR EDUCATIONAL PURPOSES ONLY, DO NOT USE IN PRODUCTION
# Source code is provided as-is without warranty, and disclaiming liability for damages resulting from using the source code.

import socket;
from struct import *;
import sys;

buf = "";
sck = socket.socket(socket.AF_INET, socket.SOCK_RAW, socket.IPPROTO_TCP);
buf = sck.recv(2048);
res = "";
cnt = 0;
print(len(buf));
for i in range(len(buf)):
  res += '{:02X} '.format(ord(buf[i]));
  cnt += 1;
  if cnt == 16:
    print(res);
    res = "";
    cnt = 0;
print(res);
